### PR TITLE
fix: issue-3425 Profiler: add test based on included datatype, issue-3407 Deleting column test shows blank in data-profiler for that column

### LINKED ISSUE #3407: Deleting column test shows blank under tests in data-profiler for that column
**Affected module**
UI

**Describe the bug**
Deleting column test shows blank under tests in data-profiler for that column 

**To Reproduce**
- Delete a column test
- Go to profiler
- Check under `tests` for the deleted column test.
- Shows blank instead of showing `No tests available`

**Expected behavior**
Should show `No tests available` when no tests are present

Screenshot 2022-03-14 at 6 13 49 PM

Screenshot 2022-03-14 at 6 18 56 PM

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**
NA

### LINKED ISSUE #3425: Profiler: add column test in profiler is only excluding "struct" datatype
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI: Profiler tab

**Describe the bug**
Currently condition is only exclude "struct" type datatype, but we have many other datatype which is not supported for test.

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**
We should modify the condition based on what datatype we support.

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

Closes #3425 Profiler: add column test in profiler is only excluding "struct" datatype
Closes #3407 Deleting column test shows blank under tests in data-profiler for that column

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/158353920-cf1a2600-e44c-4f02-92e3-227a2e1434a5.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@Sachin-chaurasiya @darth-coder00 @shahsank3t

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts [hunk 1]
```diff
@@ -17,3 +17,11 @@ export enum ConstraintTypes {
   NOT_NULL = 'NOT_NULL',
   UNIQUE = 'UNIQUE',
 }
+
+export enum SupportedDatatypeForTest {
+  varchar = 'varchar',
+  timestamp = 'timestamp',
+  date = 'date',
+  numeric = 'numeric',
+  boolean = 'boolean',
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts:26]: Enum keys should be capitalised

File: openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts [hunk 2]
```diff
@@ -17,3 +17,11 @@ export enum ConstraintTypes {
   NOT_NULL = 'NOT_NULL',
   UNIQUE = 'UNIQUE',
 }
+
+export enum SupportedDatatypeForTest {
```
Comment (darth-coder00) [inline openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts:21]:
```suggestion
export enum TestSupportedDataTypes {
```

**Review by darth-coder00 — CHANGES_REQUESTED** [review]
Minor changes needed

File: openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx [hunk 3]
```diff
@@ -633,6 +633,7 @@ const DatasetDetails: React.FC<DatasetDetailsProps> = ({
                     constraint: col.constraint as string,
                     colName: col.name,
                     colType: col.dataTypeDisplay as string,
+                    dataType: col.dataType as string,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx:636]: @ShaileshParmar11 , Why are we changing this from `dataTypeDisplay` to `dataType`?

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Minor comment.

File: openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx:636]: Hi @Sachin-chaurasiya, i have't changed anything. i have added dataType as i need the dataType of column to identify whether we support the test for particular column.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx:636]: @ShaileshParmar11, My bad.

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DatasetDetails/DatasetDetails.component.tsx:636]: No issue @Sachin-chaurasiya, can we resolve this?

**Review by Sachin-chaurasiya — APPROVED** [review]
LGTM!

File: openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts]: Updated

File: openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/enums/table.enum.ts]: Updated

**Review by darth-coder00 — APPROVED** [review]

**Pull request merged**